pints/_mcmc/_slice_hyperrectangles.py

```diff
@@ -0,0 +1,283 @@
+        self._w[self._w == 0] = 1
+        self._w = 0.1 * self._w
+
+        # Flag to turn on adaptive shrinking
+        self._adaptive = False
+
+    def ask(self):
+        """ See :meth:`SingleChainMCMC.ask()`. """
+
+        # Check ask/tell pattern
+        if self._ready_for_tell:
+            raise RuntimeError('Ask() called when expecting call to tell().')
+
+        # Initialise on first call
+        if not self._running:
+            self._running = True
+
+        # Very first iteration
+        if self._current is None:
+
+            # Ask for the log pdf of x0
+            self._ready_for_tell = True
+            return np.array(self._x0, copy=True)
+
+        # Randomly position hyperrectangle:
+        # ``H = (L_1, R_1) x ... x (L_n, R_n)``
+        if not self._hyperrectangle_positioned:
+            for i, w in enumerate(self._w):
+                u = np.random.uniform()
+                self._L[i] = self._current[i] - w * u
+                self._R[i] = self._L[i] + w
+            self._hyperrectangle_positioned = True
+
+        # Sample new proposal
+        for i in range(self._n_parameters):
+            u = np.random.uniform()
+            self._proposed[i] = (self._L[i] + u * (self._R[i] - self._L[i]))
+
+        # Send trial point for checks
+        self._ready_for_tell = True
+        return np.array(self._proposed, copy=True)
+
+    def adaptive_shrinking(self):
+        """
+        Returns True/False if adaptive shrinking is on/off.
+        """
+        return self._adaptive
+
+    def current_log_pdf(self):
+        """ See :meth:`SingleChainMCMC.current_log_pdf()`. """
+        return np.copy(self._current_log_pdf)
+
+    def current_slice_height(self):
+        """
+        Returns current height value used to define the current slice.
+        """
+        return self._current_log_y
+
+    def name(self):
+        """ See :meth:`pints.MCMCSampler.name()`. """
+        return 'Slice Sampling - Hyperrectangles'
+
+    def needs_sensitivities(self):
+        """ See :meth:`pints.MCMCSampler.needs_sensitivities()`. """
+        return True
+
+    def n_hyper_parameters(self):
+        """ See :meth:`TunableMethod.n_hyper_parameters()`. """
+        return 2
+
+    def set_adaptive_shrinking(self, adaptive):
+        """
+        Turns on/off the adaptive method for shrinking the hyperrectangle.
+        """
+        self._adaptive = bool(adaptive)
+
+    def set_hyper_parameters(self, x):
+        """
+        The hyper-parameter vector is ``[width, adaptive]``.
+        See :meth:`TunableMethod.set_hyper_parameters()`.
+        """
+        self.set_width(x[0])
+        self.set_adaptive_shrinking(x[1])
+
+    def set_width(self, w):
+        """
+        Sets the width for generating the interval. This can either
+        be a single number or an array with the same number of elements
+        as the number of variables to update.
+        """
+        if type(w) == int or float:
```

Comment:
This means `if (type(w) == int) or float)`
and `if float` == True...

use `np.isscalar(w)` instead